test/support/test_config.h

```diff
@@ -90,20 +90,6 @@
 #endif
 #endif //!defined(_ENABLE_RANGES_TESTING)
 
-// to prevent the compilation warning in clang
-#ifdef __clang__
-#define STR(X) #X
-#define CLANG_DIAGNOSTIC_PUSH                                 _Pragma(STR(clang diagnostic push))
-#define CLANG_DIAGNOSTIC_POP                                  _Pragma(STR(clang diagnostic pop))
-#define CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE _Pragma(STR(clang diagnostic ignored "-Wtautological-constant-compare"))

```

Comment:
All these warnings suppression mot required anymore because now we switched off "fast math" for all tests.
Also this suppression absent in original tests of std::complex<T>.